### ISSUE TITLE
Optimize for speed

### ISSUE THREAD

**Issue by sjfleming** [broadinstitute/CellCap#36]
Use a python profiler to look for speed bottlenecks in the code.

Figure out the GPU utilization of scVI, and make sure we try to match that at least.  Currently our GPU utilization looks on the low side.

Look for mathematical modifications and simplifications to speed up the code.

For example, use something like this to examine the `loss()`, `inference()`, and `generative()` functions.
https://github.com/pyutils/line_profiler

**Comment by sjfleming** [comment]
We can definitely get rid of the for-loop here:
https://github.com/broadinstitute/CellCap/blob/1b7e1728cb7a6adafd38ca69d351c60f01b19ac3/cellcap/model.py#L190

**Comment by sjfleming** [comment]
Check out these

https://pytorch.org/tutorials/intermediate/scaled_dot_product_attention_tutorial.html

https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html

**Comment by sjfleming** [comment]
Add this to the code maybe in some kind of debugging or developer mode (have it "off" by default)

https://pytorch-lightning.readthedocs.io/en/1.5.10/api/pytorch_lightning.callbacks.gpu_stats_monitor.html

so that you can see a log of the GPU use statistics
